server/routes/api/place.py

```diff
@@ -763,3 +766,134 @@ def placeid2dcid():
                 result[inId] = dcid
         return Response(json.dumps(result), 200, mimetype='application/json')
     abort(404, 'no valid dcids found for %s' % place_ids)
+
+
+@cache.cached(timeout=3600 * 24, query_string=True)  # Cache for one day.
+@bp.route('/ranking_chart/<path:dcid>')
+def api_ranking_chart(dcid):
+    """
+    Gets the ranking data (including ranks, placename, value, date, and source) for a given place.
+
+    Return value example:
+    {
+        "statvar":{
+            "date":"2022-03",
+            "data":[
+                {
+                    "rank":1,
+                    "value":0.1,
+                    "placeDcid":"geoId/06",
+                    "placename":"California"
+                }
+            ],
+            "numDataPoints":1,
+            "exploreUrl":"/ranking/UnemploymentRate_Person_Rural/State/country/USA?h=geoId/06&unit=%",
+            "sources":[
+                "https://www.bls.gov/lau/"
+            ]
+        }
+    }
+    """
+    result = {}
+    # Get the parent place.
+    if not dcid == EARTH_DCID:
+        parent_place_list = get_parent_place(dcid).get(dcid, [])
+        # Get the last parent place returned.
+        if parent_place_list:
+            parent_place_dcid = parent_place_list[-1]["dcid"]
+            place_type = get_place_type(dcid)
+        else:
+            return Response(json.dumps(result),
+                            200,
+                            mimetype='application/json')
+    else:
+        parent_place_dcid = EARTH_DCID
+        place_type = "Country"
+    configs = get_ranking_chart_configs()
+    # Get the first stat var of each config.
+    stat_vars, _ = shared_api.get_stat_vars(configs)
+    sv_data = dc.get_stat_set_within_place(parent_place_dcid, place_type,
+                                           list(stat_vars), "")
+    sv_data_values = sv_data.get("data", {})
+    if not sv_data or not sv_data_values:
+        return Response(json.dumps(result), 200, mimetype='application/json')
+    ## Get all the place names of dcids in the sv_data
+    place_dcids = set()
+    for value in sv_data_values.values():
+        for place_dcid in value["stat"]:
```

Comment:
nit: instead of iterating through all the place_dcids, we can do a union on the keys like: https://github.com/datacommonsorg/website/blob/master/server/routes/api/stats.py#L456-L461